File: app/src/main/java/com/example/note_app/database/RoomDAO.java
//package com.example.note_app.database;
//
//import androidx.lifecycle.MutableLiveData;
//import androidx.room.Dao;
//import androidx.room.Delete;
//import androidx.room.Insert;
//import androidx.room.Query;
//import androidx.room.Update;
//
//import com.example.note_app.home.NoteModel;
//
//import java.util.List;
//
//@Dao
//public interface RoomDAO {
//
////    @Insert
////    public void Insert(UsernamePassword... usernamePasswords);
////
////    @Update
////    public void Update(UsernamePassword... usernamePasswords);
////
////    @Delete
////    public void Delete(UsernamePassword usernamePassword);
////
////    @Query("Select * from login where usename = :username")
////    public UsernamePassword getUserwithUsername(String username);
////
////    @Query("Select * from login where isloggedIn = 1")
////    public UsernamePassword getLoggedInUser();
//
//
//    @Insert
//    public void Insert(NoteModel... noteModels);
//
//    @Update
//    public void Update(NoteModel... noteModels);
//
//    @Delete
//    public void Delete(NoteModel reminders);
//
//    @Query("Select * from note order by createdAt")
//    public List<NoteModel> orderThetable();
//
//    @Query("Select * from note Limit 1")
//    public NoteModel getRecentEnteredData();
//
//    @Query("Select * from note")
//    public  M getAll();
//
//}

File: app/src/main/java/com/example/note_app/database/RoomDAO.kt
package com.example.note_app.database

import androidx.room.*
import com.example.note_app.home.NoteModel

@Dao
interface RoomDAO {


    //    @Insert
    //    public void Insert(UsernamePassword... usernamePasswords);
    //
    //    @Update
    //    public void Update(UsernamePassword... usernamePasswords);
    //
    //    @Delete
    //    public void Delete(UsernamePassword usernamePassword);
    //
    //    @Query("Select * from login where usename = :username")
    //    public UsernamePassword getUserwithUsername(String username);
    //
    //    @Query("Select * from login where isloggedIn = 1")
    //    public UsernamePassword getLoggedInUser();
    @Insert
    fun insert(vararg noteModels: NoteModel?)

    @Update
    fun update(vararg noteModels: NoteModel?)

    @Delete
    fun delete(noteModel: NoteModel?)

    @Query("SELECT * FROM note where id = :id")
    fun getNoteById(id:Int) : NoteModel

    @Query("Select * from note order by createdAt")
    fun orderThetable(): List<NoteModel?>?

    @Query("Select * from note Limit 1")
    fun getRecentEnteredData(): NoteModel?

    @Query("Select * from note")
    fun getAll(): MutableList<NoteModel>?
}